glpi-project/glpi
# Add permission manager

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Add PermissionManager class to try and manage permissions more centrally.
For now, this only adds the missing capability of being able to check rights (the profile rights, not the actual permission checks some items do like Ticket::canUpdate) for other users.

You can:
1. Get a combined set of rights from an array of profile IDs
2. Get an array of the user's profiles that have a specific right
3. Check if a user has a specific profile
4. Check if a user has a specific right
    - This can check the current profile or all profiles
    - This can check the current entity or a specific entity

There are optimizations for checking against the current user (pass -1 as the user ID) that use the data already in the session when possible.

I don't know if there are any current uses for this in the core, but there was a plugin that I was working on in the past that needed to check permissions for another user.
